src/portal-components/DetailsPage.tsx

```diff
@@ -170,94 +148,110 @@ export default class DetailsPage extends React.Component<
     } else {
-      const synapseConfigContent = (
+      return (
         <>
           {isLoading && <BarLoader color="#878787" loading={true} height={5} />}
           {!isLoading && synapseConfigArray && (
             <DetailsPageSynapseConfigArray
+              showMenu={showMenu}
               synapseConfigArray={synapseConfigArray}
               queryResultBundle={this.state.queryResultBundle}
             />
           )}
         </>
       )
-      if (showMenu) {
-        return (
-          <div className="DetailsPage">
-            <div className="button-container">{this.renderMenu()}</div>
-            <div className="component-container" ref={this.ref}>
-              {synapseConfigContent}
-            </div>
-          </div>
-        )
-      } else {
-        return synapseConfigContent
-      }
     }
   } // end render()
+}
 
-  renderMenu = () => {
-    const { synapseConfigArray } = this.props
-    const { queryResultBundle } = this.state
-    const mapColumnHeaderToRowIndex: Dictionary<number> = {}
-    let row: string[] = []
-    if (queryResultBundle) {
-      queryResultBundle.queryResult.queryResults.headers.forEach(
-        (el, index) => {
-          mapColumnHeaderToRowIndex[el.name] = index
-        },
-      )
-      row = queryResultBundle.queryResult.queryResults.rows[0].values
-    }
-    return (
-      synapseConfigArray &&
-      synapseConfigArray.map((el: RowSynapseConfig, index) => {
-        const style: React.CSSProperties = {}
-        const { columnName = '' } = el
-        const isDisabled =
-          queryResultBundle &&
-          !row[mapColumnHeaderToRowIndex[columnName]] &&
-          !el.standalone
-        if (isDisabled) {
-          style.color = '#BBBBBC'
-          style.cursor = 'not-allowed'
-        }
-        const className = `menu-row-button ${
-          isDisabled ? '' : 'SRC-primary-background-color-hover'
-        }`
-        if (el.name === 'ExternalFileHandleLink') {
-          return (
-            <ExternalFileHandleLink
-              className={className}
-              synId={el.props.synId}
-            />
-          )
-        }
+const handleMenuClick = (
+  ref: React.RefObject<HTMLDivElement>,
+  index: number) => {
+  const wrapper = ref.current?.querySelector<HTMLDivElement>(
+    `#${COMPONENT_ID_PREFIX}${index}`,
+  )
+  if (wrapper) {
+    // https://stackoverflow.com/a/49924496
+    const offset = 85
+    const bodyRect = document.body.getBoundingClientRect().top
+    const elementRect = wrapper.getBoundingClientRect().top
+    const elementPosition = elementRect - bodyRect
+    const offsetPosition = elementPosition - offset
+    window.scrollTo({
+      top: offsetPosition,
+      behavior: 'smooth',
+    })
+  } else {
+    console.error('Could not scroll to element with index ', index)
+  }
+}
+const renderMenu = (
+  ref: React.RefObject<HTMLDivElement>,
+  synapseConfigArray?: RowSynapseConfig[],
+  queryResultBundle?: QueryResultBundle
+) => {
```

Comment:
I don't think there are any issues here since this component has no hooks, but would prefer treating this as a component since it unconditionally returns a React element, then render it with JSX instead of calling it.